### PR TITLE
Allow user to override default page size

### PR DESCRIPTION
**Pull request by MarcSabatella**
Resolves: https://github.com/musescore/MuseScore/issues/12828
Resolves: https://github.com/musescore/MuseScore/issues/12829 (partially)

MuseScore currently sets all new scores to A4 page size.  This is inappropriate in North America; the default should be Letter.  There is no great way to make this determination automatically, but MuseScore does provide a preference setting to specify a custom style file for new scores and for parts.  Unfortunately, this setting was being completely ignored for parts, and also for scores created from templates as opposed to "from scratch".

This PR fixes the bug where the custom style for parts was being completely ignored, and also allows the page size setting from the user's custom style to take precedence over that of the template, if the template was using the default page size (the template still wins if it is explicitly specifying a larger or smaller page size).  It also allows the environment variable PAPERSIZE (used on some Linux systems) to provide an additional override, and it establishes a framework to allow the default page size to be made a user-visible preference in the future, which could perhaps be set to a more reasonable locale-dependent default automatically.

See commit messages for further details.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by MarcSabatella** [comment]
The way I am approaching this is not the only way these problems could be addressed, and I'm happy to discuss further.  If someone wants to implement this using another approach instead, feel free to borrow my code.  One way or another, though, it's pretty important that those of us in North America have a way of makiing sure our scores default to Letter instead of A4.

**Comment by Tantacrul** [comment]
@RomanPudashkin @Eism - could you merge this if you are happy with the implementation. I'm unable to test it myself because I'm in the UK. I've asked a few community members to have a look and confirm it's working.

**Comment by MarcSabatella** [comment]
I've updated my PR and made my best attempt at setting the page size automatically, by checking the system locale and setting the default page size to letter if the country is any of the ones listed in the Wikipedia article as using that size.  There is no guarantee this will yield the right answer always, but at least this PR also fixes the workaround.

Here is the PR to test: https://github.com/musescore/MuseScore/issues/14498

**Comment by Jojo-Schmitz** [comment]
Seems a rebase is needed?
You may get the "without Qt" check fixed with some `#ifdef`?

**Comment by MarcSabatella** [comment]
> Seems a rebase is needed? You may get the "without Qt" check fixed with some `#ifdef`?

Rebased.  I'm happy to add ifdef around the Qt-specific code, but I'd need guidance in how to do this.  Is it just "#ifndef NO_QT_SUPPORT" as I see used a few other places?  I'll give that a shot.  The locale autodetect won't work, of course, but it will compile and still be better than the current situation, since at least the page size specified in preferences will be honored.

File: src/engraving/style/defaultstyle.cpp [hunk 1]
```diff
@@ -47,15 +48,37 @@ DefaultStyle* DefaultStyle::instance()
...
+    case QLocale::UnitedStates:
+    case QLocale::Canada:
+    case QLocale::Mexico:
+    case QLocale::Chile:
+    case QLocale::Colombia:
+    case QLocale::CostaRica:
+    case QLocale::Panama:
+    case QLocale::Guatemala:
+    case QLocale::DominicanRepublic:
+    case QLocale::Philippines:
```
Comment (Jojo-Schmitz) [inline src/engraving/style/defaultstyle.cpp:74]: Isn't it Nicaragua, El Salvador, Belize, Puerto Rico and Venzuela too?
As far as I can tell the Dominical Republic ises A4

image

Comment (MarcSabatella) [reply src/engraving/style/defaultstyle.cpp:74]: My list is based on the Wikipedia article on letter size - https://en.m.wikipedia.org/wiki/Letter_(paper_size). But I’m happy to update based on more authoritative documentation.

**Comment by cbjeukendrup** [comment]
@MarcSabatella I think that preferably the Qt stuff should be hidden at least in EngravingConfiguration (maybe even further away, but let's use EngravingConfiguration as a start). 

Perhaps you could create a `Size defaultPageSize()` method in (I)EngravingConfiguration, and hide the Qt Locale logic inside that method, and simply call the method in the DefaultStyle class. (You'll need to INJECT_STATIC `IEngravingConfiguration` in the DefaultStyle class (just like we do in the EngravingItem class.)

**Comment by MarcSabatella** [comment]
Hmm, that's getting a little beyond what I'm comfortable with - this architecture of the code still is very new to me, and I'm operating without a build system, just using the online GitHub interface to make simple changes where I can, depending on CI to build for me, etc.  But I'm happy to donate my code here to someone who wants to then make it fit the architecture better.

File: src/engraving/style/defaultstyle.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply src/engraving/style/defaultstyle.cpp:74]: Mine on the Image from https://en.wikipedia.org/wiki/Paper_size

**Pull request closed without merging**